### ISSUE TITLE
basement fails to build with GHC-7.8.4 (base < 4.8)

### ISSUE THREAD

**Issue by larskuhtz** [haskell-foundation/foundation#407]
Building `basement` with GHC-7.8.4 fails due to a missing `Bits` instance for `Natural`:

```
Basement/Types/Word128.hs:89:42:
    No instance for (Bits Natural) arising from a use of ‘.|.’
    In the expression:
      (toNatural a1 `unsafeShiftL` 64) .|. toNatural a0
    In an equation for ‘toNatural’:
        toNatural (Word128 a1 a0)
          = (toNatural a1 `unsafeShiftL` 64) .|. toNatural a0
    In the instance declaration for ‘IsNatural Word128’
```

**Comment by vincenthz** [comment]
thanks @larskuhtz for the report. I fixed it, but now that 8.2 has been released, 7.8 support is going to be patchy

**Comment by larskuhtz** [comment]
thanks @vincenthz
